lxd/storage/backend_lxd.go

```diff
@@ -3723,25 +3726,118 @@ func (b *lxdBackend) ImportBucket(projectName string, poolVol *backupConfig.Conf
 	}
 
 	// Validate config and create database entry for restored bucket.
-	bucketID, err := BucketDBCreate(context.TODO(), b, projectName, true, bucket)
+	bucketID, err := BucketDBCreate(b.state.ShutdownCtx, b, projectName, true, bucket)
 	if err != nil {
 		return nil, err
 	}
 
-	revert.Add(func() { _ = BucketDBDelete(context.TODO(), b, bucketID) })
+	revert.Add(func() { _ = BucketDBDelete(b.state.ShutdownCtx, b, bucketID) })
 
 	// Get the bucket name on storage.
-	storageBucketName := project.StorageVolume(projectName, poolVol.Bucket.Name)
+	storageBucketName := project.StorageVolume(projectName, bucket.Name)
 	storageBucket := b.GetVolume(drivers.VolumeTypeBucket, drivers.ContentTypeFS, storageBucketName, bucketConfig)
 
 	err = b.driver.ValidateVolume(storageBucket, false)
 	if err != nil {
 		return nil, err
 	}
 
+	// Recover bucket keys.
+
+	// Start minio process.
+	minioProc, err := b.ActivateBucket(projectName, bucket.Name, op)
```

Comment:
This is a problem because `ImportBucket` should (at least someday) need to work with cephobject pools that dont use minio. You can see the other invocations of b.ActivateBucket are always conditional on the bucket being memberSpecific (i.e not a remote ceph bucket).

It sounds like we need a change to ensure that ImportBucket handles (or at least cleanly passes over) recovering from a cephobject pool, and doesn't assume all buckets are minio based.

Reply:
Didnt you already add a check that skips if ErrNotSupported returned? Could you just return that in this case?

Reply:
Ah no that was in ListUnknownVolumes. I think returning an error as you propose is best for now, as it shouldn't every actually get there because it will never be in the unknown volume list.

Reply:
Yes, of course.

To clarify the check we have added previously:
In the `lxd/api_internal_recover` a function `lxd_backend#ListUnknownVolumes()` is called which calls `driver#ListVolumes()`. Since Ceph object storage does not support volumes, the recovery failed (this is currently a *bug* on master branch).